File: 0169_Majority Element/MajorityElement.py
class MajorityElement(object):
    def majorityElement(self, nums):
        current = -1
        count = 0
        for i in range(0, len(nums)):
            if count == 0:
                current = nums[i]
                count +=1
            elif current == nums[i]:
                count +=1
            else:
                count -=1
            if count > len(nums)/2:
                break
        return current

File: 0169_Majority Element/MajorityElement.cs
class Solution
{
    public int MajorityElement(int[] nums)
    {
        int current = int.MinValue;
        int count = 0;
        for(int i = 0; i < nums.Length; i++)
        {
            if(count == 0)
            {
                current = nums[i];
                count++;

            }else if(nums[i] == current)
            {
                count++;
            }
            else
            {
                count--;
            }

            if (count > nums.Length / 2) break;
        }

        return current;
    }
}

